Cython/Utility/ModuleSetupCode.c

```diff
@@ -1074,34 +1074,38 @@ static CYTHON_INLINE PyObject * __Pyx_PyDict_GetItemStrWithError(PyObject *dict,
   #define __Pyx_SET_SIZE(obj, size) Py_SIZE(obj) = (size)
 #endif
 
-#if CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS && PY_VERSION_HEX >= 0x030d00b1
+#if __PYX_LIMITED_VERSION_HEX >= 0x030d0000
```

Comment:
I'm not sure how much it matters, but I think we still prefer

```
#define __Pyx_PyList_GetItemRef(o, i) __Pyx_NewRef(PyList_GET_ITEM(o, i))
```

for Python 3.14 outside the limited API and with `CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS` disabled. It avoids a type-check and a boundscheck and I think it's mostly used in places where we're confident of both of these.

Maybe that suggests a slight naming issue since in the current version
* `__Pyx_PyDict_GetItemRef` is "as close as we can get to `PyDict_GetItem`
* `__Pyx_PyList_GetItemRef` is "as close as we can (safely) get to `__Pyx_NewRef(PyList_GET_ITEM(o, i))`"
so they aren't quite the same thing

Reply:
@da-woods Fixed. What about now? Did I get your request right?

Reply:
I think what David meant was that both list and dict are really independent. While `PyDict_GetItemRef()` is generally preferable if it's available (as in your previous version), `PyList_GetItemRef()` is not preferable and should only be used if we can't go faster.

Could you try separating the two functions to allow this different preference?

Reply:
@da-woods @scoder What about now?